### PR TITLE
add(main): new package rirc

### LINKED ISSUE #11876: [Package]: rirc
### Why do you want to add this package to Termux?

A terminal irc client in C

### Home page URL

https://rcr.io/rirc/

### Source code URL

https://github.com/rcr/rirc

### Packaging policy acknowledgement

- [X] The requested package complies with [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by T-Dynamos**
Closes #11876

### CODE REVIEW

File: packages/rirc/makefile.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- rirc/Makefile	2022-09-12 23:12:14.492293684 +0530
++++ rirc.mod/Makefile	2022-09-12 18:39:34.298457909 +0530
+@@ -1,6 +1,6 @@
+ VERSION := 0.1.6
+ 
+-PREFIX   ?= /usr/local
++PREFIX   ?=  @TERMUX_PREFIX@/local
```
Comment (Grimler91) [inline packages/rirc/makefile.patch:7]: This patch shouldn't be necessary, it uses `?=` and so should be overridden by the value we set

Comment (T-Dynamos) [reply packages/rirc/makefile.patch]: Thanks, I don't know this

**Comment by T-Dynamos** [comment]
I am seeing different compile behaviors
On local build
```
/home/tdynamos/rirc-0.1.6/lib/mbedtls-3.0.0.tar.gz...
/home/tdynamos/rirc-0.1.6/lib/mbedtls-3.0.0.tar.gz: OK
  CC    aes.c
  CC    aesni.c
  CC    aria.c
  CC    asn1parse.c
  CC    asn1write.c
  CC    base64.c
  CC    bignum.c
  CC    camellia.c
  CC    ccm.c
  CC    chacha20.c
  CC    chachapoly.c
  CC    cipher.c
  CC    cipher_wrap.c
  CC    cmac.c
  CC    ctr_drbg.c
  CC    des.c
  CC    dhm.c
  CC    ecdh.c
  CC    ecdsa.c
  CC    ecjpake.c
  CC    ecp.c
  CC    ecp_curves.c
  CC    entropy.c
  CC    entropy_poll.c
  Gen   error.c
  CC    error.c
  CC    gcm.c
  CC    hkdf.c
  CC    hmac_drbg.c
  CC    md.c
  CC    md5.c
  CC    memory_buffer_alloc.c
  CC    mps_reader.c
  CC    mps_trace.c
  CC    nist_kw.c
  CC    oid.c
  CC    padlock.c
  CC    pem.c
  CC    pk.c
  CC    pk_wrap.c
  CC    pkcs12.c
  CC    pkcs5.c
  CC    pkparse.c
  CC    pkwrite.c
  CC    platform.c
  CC    platform_util.c
  CC    poly1305.c
  CC    psa_crypto.c
  CC    psa_crypto_aead.c
  CC    psa_crypto_cipher.c
  CC    psa_crypto_client.c
  CC    psa_crypto_driver_wrappers.c
  CC    psa_crypto_ecp.c
  CC    psa_crypto_hash.c
  CC    psa_crypto_mac.c
  CC    psa_crypto_rsa.c
  CC    psa_crypto_se.c
  CC    psa_crypto_slot_management.c
  CC    psa_crypto_storage.c
  CC    psa_its_file.c
  CC    ripemd160.c
  CC    rsa.c
  CC    rsa_alt_helpers.c
  CC    sha1.c
  CC    sha256.c
  CC    sha512.c
  CC    threading.c
  CC    timing.c
  CC    version.c
  Gen   version_features.c
  CC    version_features.c
  CC    ../3rdparty/everest/library/everest.c
  CC    ../3rdparty/everest/library/x25519.c
  CC    ../3rdparty/everest/library/Hacl_Curve25519_joined.c
  AR    libmbedcrypto.a
  CC    x509.c
  CC    x509_create.c
  CC    x509_crl.c
  CC    x509_crt.c
  CC    x509_csr.c
  CC    x509write_crt.c
  CC    x509write_csr.c
  AR    libmbedx509.a
  CC    debug.c
  CC    net_sockets.c
  CC    ssl_cache.c
  CC    ssl_ciphersuites.c
  CC    ssl_cli.c
  CC    ssl_cookie.c
  CC    ssl_msg.c
  CC    ssl_srv.c
  CC    ssl_ticket.c
  CC    ssl_tls.c
  CC    ssl_tls13_keys.c
  AR    libmbedtls.a
cc -O2 -flto -DNDEBUG src/io.c
cc -O2 -flto -DNDEBUG src/rirc.c
cc -O2 -flto -DNDEBUG src/state.c
cc -O2 -flto -DNDEBUG src/utils/utils.c
cc -flto -pthread rirc
lto-wrapper: warning: using serial compilation of 5 LTRANS jobs
lto-wrapper: note: see the ‘-flto’ option documentation for more information

```
On CI
```
gperf --output-file=src/handlers/irc_recv.gperf.out src/handlers/irc_recv.gperf
/home/builder/.termux-build/rirc/src/lib/mbedtls-3.0.0.tar.gz...
gperf --output-file=src/handlers/irc_send.gperf.out src/handlers/irc_send.gperf
gperf --output-file=src/handlers/irc_ctcp.gperf.out src/handlers/irc_ctcp.gperf
cp config.def.h config.h
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/buffer.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/channel.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/input.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/ircv3.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/mode.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/server.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/components/user.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/draw.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/handlers/irc_ctcp.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/handlers/irc_recv.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/handlers/irc_send.c
aarch64-linux-android-clang  -fstack-protector-strong -Oz -DNDEBUG src/handlers/ircv3.c
src/handlers/ircv3.c:6:10: fatal error: 'mbedtls/base64.h' file not found
#include "mbedtls/base64.h"
         ^~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [Makefile:49: build/handlers/ircv3.o] Error 1
make: *** Waiting for unfinished jobs....
/home/builder/.termux-build/rirc/src/lib/mbedtls-3.0.0.tar.gz: OK
```

File: packages/rirc/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+TERMUX_PKG_VERSION=0.1.6
+TERMUX_PKG_SRCURL=https://github.com/rcr/rirc/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=95ac4fa4538d42509c5ef763afd6b85882f5b8bc0e45dced7b0cb9916961243e
+TERMUX_PKG_DEPENDS="mbedtls"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_make_install() {
+	make config.h
+	make 
+	make install
```
Comment (Grimler91) [inline packages/rirc/build.sh:14]: `make config.h` should probably be run in `termux_step_configure() instead`, seems more logical.

Should be no need to run `make` and `make install` manually, that will be done by the default `termux_step_make` and `termux_step_make_install`

**Review by Grimler91 — CHANGES_REQUESTED** [review]
Thanks for the PR! Some suggestions for improvements

**Comment by xtkoba** [comment]
Looks like abandoned. @T-Dynamos Still work in progress?

**Comment by T-Dynamos** [comment]
@xtkoba Adding -Lmbedtls to `CFLAGS` will fix the error?

**Review by PraveenNanda124 — COMMENTED** [review]
Great work

**Comment by xtkoba** [comment]
Some hacks:

* Unbundle Mbed TLS
* Define `*FLAGS` properly
* Avoid duplicate definition of `struct user`
* Fix hardcoded paths

**Pull request merged**